### ISSUE TITLE
Better DX for using twind, having IDE autocompletions OOTB with VSCode

### ISSUE THREAD

**Issue by ranyitz** [denoland/fresh#392]
After generating a new project and answering that I would like to use `twind` 

```
Do you want to use 'twind' (https://twind.dev/) for styling? [y/N] y
```

IDE auto completions doesn't work OOTB.

Considering you're a VSCode user, there are two other things that need to be added:
1. Installation of [the extension](https://github.com/tailwindlabs)
2. having a `tailwind.config.js` or `tailwind.config.cjs` ([see installation](https://github.com/tailwindlabs/tailwindcss-intellisense#installation))

To solve this, we can consider the following:
1. use the `extensions.json` file in order to recommend the tailwind extension. (another point is that we can use this file in order to recommend the `deno` extension as well)
2. generate a `tailwind.config.js` file when choosing the `--twind` option
_____________

If you'll decide that these are good changes, I'd be happy to contribute

**Comment by ranyitz** [comment]
It seems that the VSCode files are already being added on #367. This can be added on top of the same prompt (e.g. if you say that you want twind and vscode, we can add the relevant configuration files)

**Comment by lucacasonato** [comment]
I am going to re-write the styling integration in a couple of weeks. Let's do this after those changes are made (so around Fresh 1.1.0).

**Comment by ranyitz** [comment]
Thanks @lucacasonato, i'll wait for `1.1.0` and submit a PR if needed.

**Comment by RileyMShea** [comment]
A solution provided for `fresh` in the vscode twind extension repo:

[twind autocomplete doesn't work in deno environment](https://github.com/tw-in-js/vscode-twind-intellisense/issues/9#issuecomment-1173095315)

> Not sure what the best solution is, but it seems like the plugin is not activating since it can't find a twind config file. Creating a `./config/twind.config.ts` file with the following content will make it work:
> 
> ```
> /** @jsx h */
> 
> /** @type {import('twind').Configuration} */
> ```

Don't know if the styling integration update would use this or if that's diverging in another direction, but it works for me as of fresh 1.0.0 (assuming the twind extension is installed)

**Comment by ranyitz** [comment]
Yes @RileyMShea, I did the same. The idea of this issue is to create this file on project generation, including the relevant annotations to make the autocompletions works OOTB.

**Comment by timfee** [comment]
I worked around this by adding `twind.config.ts` in my project root:
```tsx
// ./twind.config.ts
import { Configuration } from 'twind';

export const config: Configuration = {
  darkMode: 'class',
  mode: 'silent',
  theme: {
    extend: {
      fontFamily: {
        'hairline': 'Bungee Hairline',
      },
    },
  },
};
```

and then importing that in `main.ts` and in `utils/twind.ts`:
```tsx
./utils/twind.ts

import { IS_BROWSER } from '$fresh/runtime.ts';
import { setup } from 'twind';
export * from 'twind';
import { config } from '../twind.config.ts';

if (IS_BROWSER) setup(config);
```
```tsx
./main.ts

// snip
import { setup } from '@twind';
import { config } from './twind.config.ts';
// snip
```

The plugin author [says](https://github.com/tw-in-js/typescript-plugin#usage) it looks in root, `public`, `src` or `config` -- seems like maybe a FR for them would be to allow us to specify custom directories to look for this file (`util`)

**Comment by marvinhagemeister** [comment]
Closing this as we've switched to Tailwind CSS which allows us to compile the style sheet ahead of time and move it out of the render path. The Tailwind CSS plugin will be released in the next Fresh 1.6 version which I'm hoping to release later this week.
